packages/types/src/extensions/retry.ts

```diff
@@ -0,0 +1,20 @@
+import { RetryStrategyV2 } from "../retry";
+import { Provider, RetryStrategy } from "../util";
+
+/**
+ * A configuration interface with methods called by runtime extension
+ * @internal
+ */
+export interface RetryStrategyConfiguration {
+  /**
+   * Set retry strategy used for all http requests
+   * @param retryStrategy
+   */
+  setRetryStrategy(retryStrategy: Provider<RetryStrategyV2 | RetryStrategy>): void;
+
+  /**
+   * Get retry strategy used for all http requests
+   * @param retryStrategy
+   */
+  retryStrategy(): Provider<RetryStrategyV2 | RetryStrategy>;
```

Comment:
nit, but since `Provider<RetryStrategyV2 | RetryStrategy>` is repeated so much, maybe make a type alias for it?